packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -147,6 +157,26 @@ class DraggableScrollableSheet extends StatefulWidget {
   /// The default value is true.
   final bool expand;
 
+  /// Whether the widget should snap between [snapSizes] when the user lifts
+  /// their finger during a drag.
+  ///
+  /// If the user's finger was still moving when they lifted it, the widget will
+  /// snap to the next snap size (see [snapSizes]) in the direction of the drag.
+  /// If their finger was still, the widget will snap to the nearest snap size.
+  final bool snap;
+
+  /// A list of target sizes that the widget should snap to.
+  ///
+  /// Snap sizes are fractional values of the parent container's height. They
+  /// must be listed in increasing order and be between [minChildSize] and
+  /// [maxChildSize].
+  ///
+  /// The [minChildSize] and [maxChildSize] are implicitly included in snap
+  /// sizes and do not need to be specified here. For example, `snapSizes = [.5]`
+  /// will result in a sheet that snaps between [minChildSize], `.5`, and
+  /// [maxChildSize].
+  final List<double>? snapSizes;
```

Comment:
What do you think about creating a specific object for this instead? 

Asking because when I read the param list I found myself wondering why there are two properties for snapping and what it means to set one but not the other. There are also some rules about what goes into the list of snap points and what kind of numbers they have to be. 

I suspect that if we created a class for this it would be a bit more self-explanatory and harder to misuse. WDYT?

Reply:
I think it's confusing to have a bool argument and a field - it's either there or not, so null or not null seems to make sense.

That said, if you wanted an `enabled` property on `SnapConfiguration` I wouldn't really complain either.

In Dart, newing up objects is pretty cheap, and with named properties I think it should be fairly idiomatic. I'm suggesting switching to a class here :)

Reply:
@dnfield bump

Reply:
@dnfield bump?

Reply:
Asserting things about what's in the list means this can't be const anymore, which may end up being breaking. 

I think if we just use a `SplayTreeSet` it should work - which is similar to what we end up doing in gradient.dart to make sure the stops and colors are sorted and unique anyway.

Reply:
(We could still do the asserts at time of use instead of time of construction of course)

Reply:
Do you mean use `SplayTreeSet<double>` as our input type for `snapPoints`?

We can't enforce that the input set is using least-to-greatest ordering can we? We could Take in an arbitrarily ordered list and then convert it to a LTG ordered `SplayTreeSet` internally. I marginally prefer a list that is asserted to be LTG ordered, but I'm fine with either.

Reply:
`SplayTreeSet<double>` will automatically be ordered least to greatest with duplicates removed.

Keeping it as a list and documenting the expectation is fine - it could be converted to a SplayTreeSet later. Having the asserts is fine too, but they can't be on the ctor without dropping const, so they'd have to be on `build` or `initState` or something like that.

Reply:
Awesome! If we're okay with a flat list and asserts, then I think we're ready to move forward as is?

`_impliedSnapSizes` asserts that the list is properly ordered and between min and max before it is used:
https://github.com/flutter/flutter/pull/84394/files#diff-8a2b7e9da235db9760f0046cbbceddc5e41c78c9285a315141c401f715bd269fR364

and there's a test that ensures the asserts are thrown on invalid input:
https://github.com/flutter/flutter/pull/84394/files#diff-6d377811be6bfec7ab69f334aca4ddbaa49baf8a6639103e86afa248919d5e75R95

Let me know if there are any other blocking issues you see!